.cypress/e2e/saml/saml_auth_test.spec.js

```diff
@@ -0,0 +1,126 @@
+/*
+ *   Copyright OpenSearch Contributors
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ALL_ACCESS_ROLE } from '../../support/constants';
+
+import samlUserRoleMapping from '../../fixtures/saml/samlUserRoleMappiing.json'
+
+before(() => {
+  cy.intercept('https://localhost:9200');
+
+  // Avoid Cypress lock onto the ipv4 range, so fake `visit()` before `request()`.
+  // See: https://github.com/cypress-io/cypress/issues/25397#issuecomment-1402556488
+  cy.visit('http://localhost:5601');
+
+  cy.createRoleMapping(ALL_ACCESS_ROLE, samlUserRoleMapping);
+  cy.clearCookies();
+  cy.clearLocalStorage();
+});
+
+afterEach(() => {
+  cy.clearCookies();
+  cy.clearLocalStorage();
+});
+
+describe('Log in via SAML', () => {
+  const samlLogin = () => {
+    cy.get('input[id=userName]', { timeout: 15000}).should('be.visible');
```

Comment:
Do we need all these inline timeouts or can we use the defaults?

Reply:
+1

Reply:
Hi @peternied and @cwperks, thanks for the feedback. According to the documentation (https://docs.cypress.io/guides/references/configuration#Timeouts), the default command (such as `cy.get()`) timeout is 4 seconds. And I was testing with the default timeout on my local machine which is totally fine. However, the github runner does require longer time for the loading (especially on the idp page), thats why I was adding these timeouts here. Otherwise, the test case may become flaky on the github runner.

Reply:
What do the values in `cypress.config.js` do?

Reply:
My bad, I forgot I have changed the default val there. Good catch! I have removed all the timeouts in tests